documentation.md

```diff
@@ -0,0 +1,521 @@
+
+### Burst support and addressing
+
+#### Burst support
+
+Bursts are not allowed to cross boundaries defined by _BurstAddrLSBs_ bits. If a burst does, then it is automatically considered a wrap burst relative to this boundary. Therefore, the simulated memory controller does not make a difference between _incr_ and _wrap_ bursts. Boundaries must not match with bank row widths.
+
+Fixed bursts, additionally, are supported by setting the _burst_fixed_ field of the slot, which will provide the same address to all the entries.
+
+#### Entry addressing
+
+Taking advantage of the boundaries described above, all the entries of a slot share all their address bits, except the _BurstAddrLSBs_ least significant bits. The per-entry address is therefore calculated with the slot address as a base, to which the burst size bits are iteratively added, using adders only on the _BurstAddrLSBs_ LSBs. The wrap modulo operation is implicitly performed when the adders overflow.
+
+<figure class="image">
+  <img src="https://i.imgur.com/H8IVMBk.png" alt="Simulated memory controller internal overview">
+  <figcaption>Fig: Individual burst entry address dynamic calculation</figcaption>
+</figure>
+
+## Future work
+
+### Delay calculator
+
+#### DRAM refreshing
+
+DRAM refreshing simulation is currently not implemented. It can be implemented by periodically setting rank counters to a large value. One has to be careful in the implementation to not interfere with the three-cycles-early _mem_done_ setting.
+
+#### Rank interleaving
+
+Currently, only one rank is implemented, but the necessary infrastructure to implement different kinds of interleaving are already integrated, notably the optimizations are implemented per-rank (visible through the _for (genvar i_rk..._ loops).
+
+It remains mostly to define how interleaving is mapped, and to add additional information to the slots in case a memory request entry is split between several ranks. This situation seems preferable to avoid, by using higher-order bits to select the rank mapping.
+
+#### Request cost precision
+
+So far, all the memory request entries with the same address have the same cost, regardless of the burst size, for instance. To implement finer-grained costs, more cost categories may be necessary, making the physical implementation potentially more expensive in terms of required area and delays.
+
+#### Scheduling strategy
+
+The implemented scheduling strategy is a mainstream and representative strategy. The delay calculator has been designed in a way that the scheduling strategy can reasonably easily be extended: the request ages are explicitly maintained; the outstanding entries are stored and accessible concurrently. This provides support for more complex scheduling strategies implementations.
```

Comment:
mainstream -> state-of-the-art? or well-known?

Reply:
Okay! I picked well-known.